src/test/ui/liveness/liveness-issue-77915.rs

```diff
@@ -0,0 +1,36 @@
+// Ensure inout asm! operands are marked as used by the liveness pass
+
+// only-x86_64
```

Comment:
This test could also be made architecture independent by doing something like this:

```
unsafe fn rep_movsb(mut src: *const u8) {
    asm!("/*{0}*/", inout(reg) src);
}
```

It would allow testing this by people who use something else than x86_64 as a host.

Reply:
Fixed in the latest changes, thanks!